### ISSUE TITLE
OpenEmuHelperApp crashes when switching to Soundflower/iShowU/Multi-Output Device

### ISSUE THREAD

**Issue by gingerbeardman** [OpenEmu/OpenEmu#4055]
Summary of issue

I use Soundflower to be able to record system audio with screen recordings and also mic input (recording gameplay with audio and mic chat) but OpenEmu doesn't work well with such configuration.

# Steps to reproduce

1. [Install & Setup Soundflower ](https://github.com/mattingalls/Soundflower/releases)
1. Use OpenEmu with Built-In sound
1. Change output to Soudnflower device or Multi-Output Device, using Audio MIDI Setup or System Preferences > Sound > Output

# Expected Behavior

- Sound should comtinue

# Observed Behavior

- OpenEmuHelperApp hangs/crashes

# Debugging Information

```log
default	14:49:30.703626 +0000	OpenEmuHelperApp	AVAudioIONodeImpl.mm:706:_GetHWFormat: Unexpected number of channels in asbd: agg device (236) has 0, input sub-device (90) has 2 channels. Will query the formats again!
```

- OpenEmu Version: 2.1.1 290-g98cdf95-Release
- macOS Version: 10.14.6 (18G1012)

**Comment by TonysKindaWeird** [comment]
I'm having the same issue since I updated my OpenEmu software. I haven't updated in a very long time and finally did only to now have this issue.

**Comment by kevinoppa** [comment]
> Summary of issue
> 
> I use Soundflower to be able to record system audio with screen recordings and also mic input (recording gameplay with audio and mic chat) but OpenEmu doesn't work well with such configuration.
> 
> # Steps to reproduce
> 1. [Install & Setup Soundflower ](https://github.com/mattingalls/Soundflower/releases)
> 2. Use OpenEmu with Built-In sound
> 3. Change output to Soudnflower device or Multi-Output Device, using Audio MIDI Setup or System Preferences > Sound > Output
> 
> # Expected Behavior
> * Sound should comtinue
> 
> # Observed Behavior
> * OpenEmuHelperApp hangs/crashes
> 
> # Debugging Information
> ```
> default	14:49:30.703626 +0000	OpenEmuHelperApp	AVAudioIONodeImpl.mm:706:_GetHWFormat: Unexpected number of channels in asbd: agg device (236) has 0, input sub-device (90) has 2 channels. Will query the formats again!
> ```
> 
> * OpenEmu Version: 2.1.1 290-g98cdf95-Release
> * macOS Version: 10.14.6 (18G1012)

after reading your post, i started messing around with my midi setup.   and i finally got my openemu to work.   if you still have problem with openemu, have it look like mine maybe that'll help.[
![Screen Shot 2020-02-29 at 10 14 02 PM](https://user-images.githubusercontent.com/22992221/75619415-0ef28c80-5b41-11ea-9420-5c7168197f75.png)
](url)

**Comment by kevinoppa** [comment]
> I'm having the same issue since I updated my OpenEmu software. I haven't updated in a very long time and finally did only to now have this issue.

i finally got my openemu to work... it was the midi  audio setup that cause it to freeze/crashed.
this is how i got my openemu to work...   by fixing my midi like this.   try to have your midi setup look like mine. maybe that'll help.
![Screen Shot 2020-02-29 at 10 14 02 PM](https://user-images.githubusercontent.com/22992221/75619448-71e42380-5b41-11ea-937f-c49fd973965f.png)

**Comment by SuperbFrog** [comment]
So what you changed was selected a third device in the Multi Output? That isn't working for me and is still causing OpenEmu to not load anything if Multi-Output is selected as the audio device.

**Comment by gingerbeardman** [comment]
My Audio MIDI Setup was already like that, still with the problem.

**Comment by victor-tafro** [comment]
I would like to use openemu, but forced to use sixtyforce until this is fixed.

**Comment by dvessel** [comment]
He most likely switched to multi-out while OpenEmu was actively emulating. That’s the only way it’ll work. Have the multi-output device as your sound output before opening a game and it’ll complain.

There’s a workaround by using AU Lab and routing things manually but it’s a fragile old program and straight up confusing.

**Comment by clobber** [comment]
Resolved in OpenEmu 2.3

**Comment by jasonlightner** [comment]
I'm experiencing this issue running OpenEmu 2.3.3 under macOS 10.14.6.

Console log is as follows:
`default	00:41:16.056987 -0500	OpenEmuHelperApp	AVAudioIONodeImpl.mm:706:_GetHWFormat: Unexpected number of channels in asbd: agg device (86) has 0, input sub-device (40) has 2 channels. Will query the formats again!
`

Screenshot of my Audio MIDI Setup:
<img width="743" alt="Screen Shot 2021-01-28 at 12 43 15 AM" src="https://user-images.githubusercontent.com/4294575/106095513-ecc30f00-6101-11eb-93b9-ce71e67e1a4a.png">

Any suggestions or guidance is appreciated. Thank you so much!
